sis_scraper/main.py

```diff
@@ -416,30 +421,38 @@ def optimize_column_ordering(data, num_columns=3):
                                         time["timeStart"] <= hour + minute
                                         and time["timeEnd"] > hour + minute
                                     ):
-                                        minute_idx = int(minute / 10)
-                                        hour_idx = (
-                                            int(hour / 100) - 7
-                                        )  # we start at 7am
-                                        conflict[
+                                        minute_idx = minute // 10
+                                        hour_idx = (hour - TIME_START) // 100
+                                        index = (
                                             day_offsets[day]
                                             + hour_idx * (60 // MINUTE_GRANULARITY)
                                             + minute_idx
-                                        ] = 1
+                                        )
+                                        conflict[index] = 1
+                                        sem_conflict_table[index].append(section["crn"])
 
-                    conflicts[section["crn"]] = "".join(str(e) for e in conflict)
-        # If 0 courses, or exactly one course, occupies a slot in the conflict bitstring, we can safely remove it to save space
-        # as that implies no conflicts are possible in that block.
+                    conflicts[section["crn"]] = conflict
+        # Compute unnecessary conflict bits - where a bit is defined as unnecessary if its removal does not affect the result conflict checking
         # The following code computes a list of candidates that fit this criteria
-        unnecessary_indices = [bit_index for bit_index in range(0, BIT_VEC_SIZE)
-                if sum(int(conflicts[section_crn][bit_index]) for section_crn in conflicts) <= 1]
+        unnecessary_indices = set()
+
+        for index1 in range(BIT_VEC_SIZE):
+            for index2 in range(index1 + 1, BIT_VEC_SIZE):
+                if sem_conflict_table[index1] == sem_conflict_table[index2]:
```

Comment:
How long do these loops take to run?  Is it worth checking if `index2` is already in `index1` to avoid this comparison?

Reply:
It runs pretty fast, but I implemented this in the latest push